### PR TITLE
Add a default gateway for the klarna plugin/gateway `klarna-checkout-for-woocommerce`

### PR DESCRIPTION
**Pull request by samueljseay**
As part of woocommerce/woocommerce#32258 this adds the default gateway config for Klarna's `klarna-checkout-for-woocommerce` plugin/gateway.

Alongside some modifications to that plugin, this can be used to configure required production config for Klarna. Klarna supports test/prod keys but for simplicity I have just added support for the prod keys depending on whether you're in an EU store or not.

### Detailed test instructions:

1. Checkout my branch: https://github.com/samueljseay/klarna-checkout-for-woocommerce/pull/1
2. Create a zip of the branch and install this plugin to your test site
3. Go to payment setup from the task list on a site where you've opted out of tracking and chosen either an EU country or US country (you could test both cases if you like but thats more related to the klarna plugin PR)
4. You should see the new Klarna payment method available
5. Click "setup" button for Klarna
6. It should see that the plugin is installed, then provide you the option to enter api username and secret.
7. Fill in these fields with any random values and submit the form
8. Klarna should appear in configured gateways list, clicking manage, and opening the right "Api credentials" menu in Klarna settings should show that the settings have been persisted in the plugin.

cc @woocommerce/gezora 

Any review on the PR in my fork would be welcome too obviously, I will then submit the PR to the Klarna repository.

No changelog required.

### CODE REVIEW

**Comment by jeffstieler** [comment]
> chosen either an EU country or US country

Should this be Sweden, Finland, or Norway instead?

https://github.com/woocommerce/woocommerce-admin/pull/7085/files#diff-c1526e4249293913199bb0a1bcc13e1c74d76156e2cfdc66c3d2625d0e1e2f87R68

File: src/Features/RemotePaymentMethods/DefaultPaymentGateways.php [hunk 1]
```diff
@@ -58,6 +58,17 @@ public static function get_all() {
...
 				),
 			),
+			array(
+				'key'        => 'kco',
+				'title'      => __( 'Klarna', 'woocommerce-admin' ),
+				'content'    => __( 'Choose the payment that you want, pay now, pay later or slice it. No credit card numbers, no passwords, no worries.', 'woocommerce-admin' ),
+				'image'      => WC()->plugin_url() . '/assets/images/klarna-black.png',
+				'plugins'    => array( 'klarna-checkout-for-woocommerce' ),
+				'is_visible' => array(
+					self::get_rules_for_countries( array( 'SE', 'FI', 'NO' ) ),
```
Comment (jeffstieler) [inline src/Features/RemotePaymentMethods/DefaultPaymentGateways.php:68]: Is this supposed to be looking for the US or EU instead?

**Review by jeffstieler — DISMISSED** [review]
I was able to test the flow once I specified either Sweden, Norway or Finland, but then the credentials ended up in the "US" settings.

File: src/Features/RemotePaymentMethods/DefaultPaymentGateways.php (hunk 1 above)
Comment (samueljseay) [reply src/Features/RemotePaymentMethods/DefaultPaymentGateways.php:68]: @jeffstieler its not, these fallback gateways are the ones defined in the original payment methods list in JS (`getPaymentMethods`) so the "is visible" rules are copied from there.

When using the "is eu" check that they wrote I assumed it was gonna check for a european country and in other cases place the settings in US. I thought I had seen this working in development but I'll have to take another look. I'm definitely making some assumptions about the code in their plugin. I might have to see if there is a better approach.

Comment (joshuatf) [reply src/Features/RemotePaymentMethods/DefaultPaymentGateways.php:68]: Good question.  I see more countries available for Klarna, but guess we're limiting to these 3 to show specific gateways.  These are the current used for Klarna in the task list, so I'm inclined to approve, but cc @pmcpinto for any background on why we only use these 3 countries.

**Review by joshuatf — APPROVED** [review]
Looks good, @samueljseay!  Left a few comments in https://github.com/samueljseay/klarna-checkout-for-woocommerce/pull/1#pullrequestreview-674683018 that I think addresses the issue that @jeffstieler noted.

File: src/Features/RemotePaymentMethods/DefaultPaymentGateways.php (hunk 1 above)
Comment (pmcpinto) [reply src/Features/RemotePaymentMethods/DefaultPaymentGateways.php:68]: >  I see more countries available for Klarna, but guess we're limiting to these 3 to show specific gateways. These are the current used for Klarna in the task list, so I'm inclined to approve, but cc @pmcpinto for any background on why we only use these 3 countries.

We promote two different Klarna extensions: Checkout and Payments. Checkout should be displayed in 3 countries and Payments in 11. More info here: #5838

**Pull request merged**